javalanglib/src/main/scala/java/lang/_String.scala

```diff
@@ -319,14 +322,326 @@ final class _String private () // scalastyle:ignore
+   * reference file at
+   *
+   *   https://unicode.org/Public/13.0.0/ucd/SpecialCasing.txt
+   *
+   * That file first contains a bunch of locale-insensitive special cases,
+   * which we do not need to handle. Only the last two sections about locale-
+   * sensitive special-cases are important for us.
+   *
+   * Some of the rules are further context-sensitive, using predicates that are
+   * defined in Section 3.13 "Default Case Algorithms" of the Unicode Standard,
+   * available at
+   *
+   *   http://www.unicode.org/versions/Unicode13.0.0/
+   *
+   * We based the implementations on Unicode 13.0.0. It is worth noting that
+   * there has been no non-comment changes in the SpecialCasing.txt file
+   * between Unicode 4.1.0 and 13.0.0 (perhaps even earlier; the version 4.1.0
+   * is the earliest that is easily accessible).
+   */
+
+  def toLowerCase(locale: Locale): String = {
+    locale.getLanguage() match {
+      case "lt"        => toLowerCaseLithuanian()
+      case "tr" | "az" => toLowerCaseTurkishAndAzeri()
+      case _           => toLowerCase()
+    }
+  }
+
+  private def toLowerCaseLithuanian(): String = {
+    /* Relevant excerpt from SpecialCasing.txt
+     *
+     * # Lithuanian
+     *
+     * # Lithuanian retains the dot in a lowercase i when followed by accents.
+     *
+     * [...]
+     *
+     * # Introduce an explicit dot above when lowercasing capital I's and J's
+     * # whenever there are more accents above.
+     * # (of the accents used in Lithuanian: grave, acute, tilde above, and ogonek)
+     *
+     * 0049; 0069 0307; 0049; 0049; lt More_Above; # LATIN CAPITAL LETTER I
+     * 004A; 006A 0307; 004A; 004A; lt More_Above; # LATIN CAPITAL LETTER J
+     * 012E; 012F 0307; 012E; 012E; lt More_Above; # LATIN CAPITAL LETTER I WITH OGONEK
+     * 00CC; 0069 0307 0300; 00CC; 00CC; lt; # LATIN CAPITAL LETTER I WITH GRAVE
+     * 00CD; 0069 0307 0301; 00CD; 00CD; lt; # LATIN CAPITAL LETTER I WITH ACUTE
+     * 0128; 0069 0307 0303; 0128; 0128; lt; # LATIN CAPITAL LETTER I WITH TILDE
+     */
+
+    /* Tests whether we are in an `More_Above` context.
+     * From Table 3.17 in the Unicode standard:
+     * - Description: C is followed by a character of combining class
+     *   230 (Above) with no intervening character of combining class 0 or
+     *   230 (Above).
+     * - Regex, after C: [^\p{ccc=230}\p{ccc=0}]*[\p{ccc=230}]
+     */
+    def moreAbove(i: Int): scala.Boolean = {
+      import Character._
+      val len = length()
+
+      @tailrec def loop(j: Int): scala.Boolean = {
+        if (j == len) {
+          false
+        } else {
+          val cp = this.codePointAt(j)
+          combiningClassNoneOrAboveOrOther(cp) match {
+            case CombiningClassIsNone  => false
+            case CombiningClassIsAbove => true
+            case _                     => loop(j + charCount(cp))
+          }
+        }
+      }
+
+      loop(i +  1)
```

Comment:
There is an extraneous space I missed before the 1.

Reply:
Good catch. Fixed.